x-pack/plugins/case/server/client/client.ts

```diff
@@ -59,96 +61,190 @@ export class CaseClientHandler implements CaseClient {
+        cases,
+        caseClient: this,
+        logger: this.logger,
+      });
+    } catch (error) {
+      const caseIDVersions = cases.cases.map((caseInfo) => ({
+        id: caseInfo.id,
+        version: caseInfo.version,
+      }));
+      throw createCaseError({
+        message: `Failed to update cases using client: ${JSON.stringify(caseIDVersions)}: ${error}`,
+        error,
+        logger: this.logger,
+      });
+    }
   }
 
   public async addComment({ caseId, comment }: CaseClientAddComment) {
-    return addComment({
-      savedObjectsClient: this._savedObjectsClient,
-      caseService: this._caseService,
-      userActionService: this._userActionService,
-      caseClient: this,
-      caseId,
-      comment,
-      user: this.user,
-    });
+    try {
+      return addComment({
+        savedObjectsClient: this._savedObjectsClient,
+        caseService: this._caseService,
+        userActionService: this._userActionService,
+        caseClient: this,
+        caseId,
+        comment,
+        user: this.user,
+        logger: this.logger,
+      });
+    } catch (error) {
+      throw createCaseError({
+        message: `Failed to add comment using client case id: ${caseId}: ${error}`,
+        error,
+        logger: this.logger,
+      });
+    }
   }
 
   public async getFields(fields: ConfigureFields) {
-    return getFields(fields);
+    try {
+      return getFields(fields);
+    } catch (error) {
+      throw createCaseError({
+        message: `Failed to retrieve fields using client: ${error}`,
+        error,
+        logger: this.logger,
+      });
+    }
   }
 
   public async getMappings(args: MappingsClient) {
-    return getMappings({
-      ...args,
-      savedObjectsClient: this._savedObjectsClient,
-      connectorMappingsService: this._connectorMappingsService,
-      caseClient: this,
-    });
+    try {
+      return getMappings({
+        ...args,
+        savedObjectsClient: this._savedObjectsClient,
+        connectorMappingsService: this._connectorMappingsService,
+        caseClient: this,
+        logger: this.logger,
+      });
+    } catch (error) {
+      throw createCaseError({
+        message: `Failed to get mappings using client: ${error}`,
+        error,
+        logger: this.logger,
+      });
+    }
   }
 
   public async updateAlertsStatus(args: CaseClientUpdateAlertsStatus) {
-    return updateAlertsStatus({
-      ...args,
-      alertsService: this._alertsService,
-      scopedClusterClient: this._scopedClusterClient,
-    });
+    try {
+      return updateAlertsStatus({
+        ...args,
+        alertsService: this._alertsService,
+        scopedClusterClient: this._scopedClusterClient,
+        logger: this.logger,
+      });
+    } catch (error) {
+      throw createCaseError({
+        message: `Failed to update alerts status using client ids: ${JSON.stringify(
+          args.ids
+        )} indices: ${JSON.stringify([...args.indices])} status: ${args.status} ${error}`,
```

Comment:
More of a personal preference here. Could help readability depending on the size of any of these fields to include line breaks between the output string, so
```
`Failed to update alerts status using client ids: ${JSON.stringify(args.ids)} \n
 indices: ${JSON.stringify([...args.indices])} \n
 status: ${args.status} ${error}`
```